api/tests/unit/app_analytics/test_analytics_db_service.py

```diff
@@ -308,3 +313,106 @@ def test_get_feature_evaluation_data_calls_get_feature_evaluation_data_from_loca
     mocked_get_feature_evaluation_data_from_local_db.assert_called_once_with(
         feature=feature, environment_id=environment.id, period=30
     )
+
+
+@pytest.mark.parametrize(
+    "period",
+    [
+        "current_billing_period",
+        "previous_billing_period",
+    ],
+)
+def test_get_usage_data_returns_empty_list_when_unset_subscription_information_cache(
+    period: str,
+    mocker: MockerFixture,
+    settings: SettingsWrapper,
+    organisation: Organisation,
+) -> None:
+    # Given
+    settings.USE_POSTGRES_FOR_ANALYTICS = True
+    mocked_get_usage_data_from_local_db = mocker.patch(
+        "app_analytics.analytics_db_service.get_usage_data_from_local_db", autospec=True
+    )
+    assert getattr(organisation, "subscription_information_cache", None) is None
+
+    # When
+    usage_data = get_usage_data(organisation, period=period)
+
+    # Then
+    assert usage_data == []
+    mocked_get_usage_data_from_local_db.assert_not_called()
+
+
+@pytest.mark.freeze_time("2023-01-19T09:09:47.325132+00:00")
+def test_get_usage_data_returns_get_usage_data_from_local_db_with_set_period_starts_at_with_current_billing_period(
+    mocker: MockerFixture, settings: SettingsWrapper, organisation: Organisation
+) -> None:
+    # Given
+    period: str = "current_billing_period"
+    settings.USE_POSTGRES_FOR_ANALYTICS = True
+    mocked_get_usage_data_from_local_db = mocker.patch(
+        "app_analytics.analytics_db_service.get_usage_data_from_local_db", autospec=True
+    )
+    now = timezone.now()
+
+    cache = OrganisationSubscriptionInformationCache.objects.create(
+        organisation=organisation,
+        current_billing_term_starts_at=now - timedelta(days=20),
+        current_billing_term_ends_at=now + timedelta(days=10),
+        api_calls_24h=2000,
+        api_calls_7d=12000,
+        api_calls_30d=38000,
+        allowed_seats=5,
+        allowed_30d_api_calls=40000,
+    )
+    assert getattr(organisation, "subscription_information_cache", None) == cache
+
+    # When
+    get_usage_data(organisation, period=period)
+
+    # Then
+    mocked_get_usage_data_from_local_db.assert_called_once_with(
+        organisation=organisation,
+        environment_id=None,
+        project_id=None,
+        date_start=datetime(2022, 12, 30, 9, 9, 47, 325132, tzinfo=timezone.utc),
+        date_stop=datetime(2023, 1, 19, 9, 9, 47, 325132, tzinfo=timezone.utc),
+    )
+
+
+@pytest.mark.freeze_time("2023-01-19T09:09:47.325132+00:00")
+def test_get_usage_data_returns_get_usage_data_from_local_db_with_set_period_starts_at_with_previous_billing_period(
+    mocker: MockerFixture, settings: SettingsWrapper, organisation: Organisation
+) -> None:
+    # Given
+    period: str = "previous_billing_period"
+
+    settings.USE_POSTGRES_FOR_ANALYTICS = True
```

Comment:
nit: Could this be just one test with parametrised `period`, `date_start` and `date_stop` arguments?

Reply:
I don't really like over parametrizing tests with the mocked assertion having to be added into the parametrized decorator. The downside of duplicated code between tests is small and I'd rather more easily read them and debug them. Not to say I never parametrize tests, I do, I just don't like the really convoluted stuff.

Reply:
How about factoring out the `OrganisationSubscriptionInformationCache` object to a common fixture then?

Reply:
That sounds like a good idea. Implementing now.